Code/C++/Climbing_Stairs.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+	Problem Description :
+		You are climbing a staircase. It takes n steps to reach the top.
+
+		Each time you can either climb 1 or 2 steps. 
+		In how many distinct ways can you climb to the top?
+	Algorithm :
+		Solving Using Dynamic Programming Approach.
+		Sub-problem Formula of each trail : steps[i] = steps[i - 1] + steps[i - 2]
+		Extra Storage Size : No extra storage Required.
+*/
+
+
+#include <iostream>
+using namespace std;
+
+long long ClimbStairs(int N) 
+{
+	
+	// Base Case : at the first stair (steps[0]) the number of ways to get there is 1
+	// Base Case : at the second stair (steps[1]) the number of ways to get there is 2
+	if(N == 1)
+		return 1;
+
+	long long s1 = 1;
+	long long s2 = 2;
+
+	// Calculate the steps at the (i - 1)th stair according to this formula :
+	// steps[i] = steps[i - 1] + steps[i - 2]
+	
+	// starting from the third stair, calculate the number of ways to reach stair i according to the previous formula 
+	for(int i = 2; i < N; i++)
+	{
+		// calculate s2 and update s1 to be equal the previous s2
+		long long tmp = s2;
+		s2 = s1 + s2;
+		s1 = tmp;		
+	}		
+
+	//Return the steps at the final stair.
+	return s2;
+}
+
+int main() 
+{
+
```

Comment:
Remove extra line here

Reply:
I'm trying to resolve the conflict but something is wrong, can you resolve it please or tell me what to do ?